### ISSUE TITLE
aws_iam_role:  diffs didn't match during apply error

### ISSUE THREAD

**Issue by chrisminton** [hashicorp/terraform#10641]
### Terraform Version
0.7.13

### Affected Resource(s)
- aws_iam_role

### Terraform Configuration Files
```hcl
data "aws_iam_policy_document" "consul_role_policy" {
  statement {
    actions                  = [
      "ec2:CreateTags",
      "ec2:DescribeTags",
      "ec2:DescribeInstances",
      "cloudwatch:PutMetricData",
    ]
    resources                = [
      "*",
    ]
  }

  statement {
    actions                  = [
      "s3:GetObject",
      "s3:ListBucket",
      "s3:GetObjectAcl",
      "s3:GetBucketAcl",
    ]
    resources                = [
      "arn:aws:s3:::*blah",
      "arn:aws:s3:::*blah/*",
    ]
  }

  statement {
    actions                  = [
      "route53:ChangeResourceRecordSets",
    ]
    resources                = [
      "arn:aws:route53:::hostedzone/XXXXXXXX",

    ]
  }

  statement {
    actions                  = [
      "route53:GetHostedZone",
      "route53:ListResourceRecordSets",
      "route53:ListHostedZones",
    ]
    resources                = [
      "*",
    ]
  }
}

data "aws_iam_policy_document" "consul_role" {
  statement {
        sid                  = ""
        actions              = [
          "sts:AssumeRole",
        ]

        principals {
          type = "Service"
          identifiers = ["ec2.amazonaws.com"]
        }
    }
}

resource "aws_iam_role_policy" "consul_role_policy" {
  name   = "consul_role_policy"
  role   = "${aws_iam_role.consul_role.id}"
  policy = "${data.aws_iam_policy_document.consul_role_policy.json}"
}

resource "aws_iam_role" "consul_role" {
  name               = "consul_role"
  assume_role_policy = "${data.aws_iam_policy_document.consul_role.json}"
}

resource "aws_iam_instance_profile" "consul_instance_profile" {
  name  = "consul_instance_profile"
  roles = ["${aws_iam_role.consul_role.name}"]
}
```

### Debug Output
https://gist.github.com/chrisminton/e1a9477269e4e21b021e7b38e40cb573

### Expected Behavior
aws_iam_role should have been destroyed and a new one created.

### Actual Behavior
`terraform apply` errored with `diffs didn't match during apply`.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. use above config, but with inline policies rather than data resources
2. `terraform apply`
3. change inline policies to data resources (as above) and change name of `aws_iam_role` and `aws_iam_instance_profile`
4. `terraform apply`
Note: `terraform plan` shows ` 4 to add, 1 to change, 4 to destroy.`

**Comment by chrisminton** [comment]
Seems this could be a duplicate of https://github.com/hashicorp/terraform/issues/2761

**Comment by mitchellh** [comment]
This is fixed in 0.8! :) To be released next week.
